### ISSUE TITLE
Saving a file with the JSON and Prettier extensions enabled causes Nova 8 to crash

### ISSUE THREAD

**Issue by reykjalin** [alexanderweiss/nova-prettier#82]
I reported this issue in https://github.com/apexskier/nova-json-language-server/issues/176 and think it'd be best to keep the issue report in one place, so I don't want to copy it all in here.

The gist is: this is probably a problem with Nova itself. I'll report the problem to the Nova developers and comment on the issue I mentioned above with updates.

Thought I'd report this here just so you're aware of the problem 🙂

**Comment by alexanderweiss** [comment]
Interesting! Thanks for following up and good to figure out if anything needs to be done to Prettier.

**Comment by WillemSleegers** [comment]
Same for TypeScript files.

**Comment by zaceno** [comment]
I have had this issue as well. As far as I can tell it happens when the "format-on-save" option is enabled.

 I'm just guessing here but I would suspect it somehow conflicts with Nova 8's new "auto-save on focus-shift" feature

**Comment by edwardloveall** [comment]
I can say that for me I disabled all extensions except prettier, then I tried to save a file and Nova 8 crashed.

**Comment by edwardloveall** [comment]
I may have spoke too soon. The file I was saving was a markdown file, which prettier checks for formatting. But then, I saved a Ruby file with the [Rubocop extension](https://github.com/jsmecham/nova-rubocop) enabled and it also crashed (even with prettier disabled). I then tried to save a Crystal file with [my own Crystal extension](https://github.com/edwardloveall/Crystal.novaextension) and it Crashed. Something about formatting or linting on save is broken on the Nova side, me thinks.
